### ISSUE TITLE
Admonitions in list items have extra margin-top on Chrome and Edge

### ISSUE THREAD

**Issue by iBug** [squidfunk/mkdocs-material#3644]
### Contribution guidelines

- [X] I've read the [contribution guidelines](https://github.com/squidfunk/mkdocs-material/blob/master/CONTRIBUTING.md) and wholeheartedly agree

### I've found a bug and checked that ...

- [X] ... the problem doesn't occur with the `mkdocs` or `readthedocs` themes
- [X] ... the problem persists when all overrides are removed, i.e. `custom_dir`, `extra_javascript` and `extra_css`
- [X] ... the documentation does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem

### Description

As the title says.

According to browser's F12 Developer Tools, this is because [this CSS selector](https://github.com/squidfunk/mkdocs-material/blob/8.2.3/src/assets/stylesheets/main/_typeset.scss#L364):

```css
.md-typeset ol li :-webkit-any(p,blockquote),
.md-typeset ul li :-webkit-any(p,blockquote) {
  margin: .5em 0;
  /* In the expanded result, only margin-top is left un-overridden */
}
```

has a higher specificity than [this CSS selector](https://github.com/squidfunk/mkdocs-material/blob/8.2.3/src/assets/stylesheets/main/extensions/markdown/_admonition.scss#L117):

```css
.md-typeset :is(.admonition-title,summary) {
  margin-bottom: 0;
  margin-top: 0;
}
```

> ![image](https://user-images.githubusercontent.com/7273074/156213932-8a916b80-e8fb-4f31-9bfd-051714a072b4.png)

Theoretically the latter has 2 class-type selectors while the former has only one, the Computed result indicates that the browser considered them "equivalent" (otherwise they would've been sorted, see [Firefox results](https://user-images.githubusercontent.com/7273074/156214931-a19463ea-28f7-4b05-8743-9875f0f5d295.png)).

### Expected behaviour

![image](https://user-images.githubusercontent.com/7273074/156209630-bab95097-95d3-4614-a19b-839dbf82a11a.png)

### Actual behaviour

![image](https://user-images.githubusercontent.com/7273074/156209561-d1236730-509b-47c9-b101-12da40843e2c.png)

### Steps to reproduce

```markdown
- List item

    !!! note "Problematic admonition"
        Content

!!! success "Working fine"
```

### Package versions

mkdocs-1.2.3, mkdocs-material-8.2.3

### Configuration

```yaml
site_name: My Docs
site_url: https://www.example.com
```

### System information

Browser:

- Microsoft Edge 98.0.1108.62 (Official build) (64-bit) on Windows 10
- Google Chrome 98.0.4758.102 (Official Build) (64-bit) on Windows 10 + Ubuntu 20.04
- Safari 15.3 (15612.4.9.1.8, 15612) on macOS Catalina

Firefox is unaffected. The selector specificity seems right.

**Comment by iBug** [comment]
It seems like `:-webkit-any()` is considered a class-type selector and has the same specificity as `.cls`, `[attr="value"]` and `:hover`. When I changed

    .md-typeset ul li :-webkit-any(p,blockquote)

into

    .md-typeset ul li :is(p,blockquote)

it immediately went down in the list (specificity recalculated) and the display result went as expected.

**Comment by iBug** [comment]
Further tests using this HTML:

```html
<style>
:is(.cls)         { color: blue; }
:is(div)          { color: red;  }
:-webkit-any(div) { color: lime; }
</style>
<div class="cls">Text</div>
```

Edge and Chrome both give lime text (wrong), while Firefox gives blue (correct). Presumably this is because FF doesn't have `:-webkit-any`.

According to [Can I Use?](https://caniuse.com/css-matches-pseudo), a considerable proportion (around 3%) of browsers only supports the prefixed `any()`, so I'm tempted to keep the buggy `:-webkit-any` and find another way around, like adding `.admonition` before `.admonition-title` to bump its specificity.

**Comment by squidfunk** [comment]
Thanks for reporting. A PR is appreciated. I think we should remove the `:is` selector on the `.md-typeset` selector causing the issue, replacing it with simple nested SCSS rules.

**Comment by iBug** [comment]
@squidfunk May I ask what was the original reason to use `:is()` than relying on SCSS to generate comma-separated selector lists?

**Comment by squidfunk** [comment]
Released as part of 8.2.4.
